module/zfs/dmu_objset.c

```diff
@@ -827,6 +828,9 @@ dmu_objset_create_impl(spa_t *spa, dsl_dataset_t *ds, blkptr_t *bp,
 	os->os_phys->os_type = type;
 	if (dmu_objset_userused_enabled(os)) {
 		os->os_phys->os_flags |= OBJSET_FLAG_USERACCOUNTING_COMPLETE;
+		if (dmu_objset_userdnused_enabled(os))
+			os->os_phys->os_flags |=
+				OBJSET_FLAG_USERDNACCOUNTING_COMPLETE;
```

Comment:
FYI, cstyle here is wrong for illumos.  Not sure if you care for ZoL.  it should be:

<2 tabs>if (...) {
<3 tabs>os->os_phys->os_flags |=
<3 tabs + 4 space>OBJSET_...;
<2 tabs>}